blocks/library/latest-comments/index.js

```diff
@@ -0,0 +1,209 @@
+			const { setAttributes } = this.props;
+
+			setAttributes( { displayExcerpt: ! displayExcerpt } );
+		}
+
+		toggleDisplayTimestamp() {
+			const { displayTimestamp } = this.props.attributes;
+			const { setAttributes } = this.props;
+
+			setAttributes( { displayTimestamp: ! displayTimestamp } );
+		}
+
+		changeCommentsToShow( commentsToShow ) {
+			const { setAttributes } = this.props;
+
+			setAttributes( { commentsToShow: parseInt( commentsToShow, 10 ) || 0 } );
+		}
+
+		render() {
+			const { latestComments } = this.state;
+			const { setAttributes } = this.props;
+
+			if ( ! latestComments.length ) {
+				return (
+					<Placeholder
+						icon="admin-post"
+						label={ __( 'Latest Comments' ) }
+					>
+						<Spinner />
+					</Placeholder>
+				);
+			}
+
+			const { focus } = this.props;
+			const { align, displayAvatar, displayTimestamp, displayExcerpt } = this.props.attributes;
+
+			return [
+				focus && (
+					<BlockControls key="controls">
+						<BlockAlignmentToolbar
+							value={ align }
+							onChange={ ( nextAlign ) => {
+								setAttributes( { align: nextAlign } );
+							} }
+							controls={ [ 'left', 'center', 'right', 'wide', 'full' ] }
+						/>
+					</BlockControls>
+				),
+				focus && (
+					<InspectorControls key="inspector">
+						<BlockDescription>
+							<p>{ __( 'Shows a list of your site\'s most recent comments.' ) }</p>
+						</BlockDescription>
+						<h3>{ __( 'Latest Comments Settings' ) }</h3>
+
+						<ToggleControl
+							label={ __( 'Display avatar' ) }
+							checked={ displayAvatar }
+							onChange={ this.toggleDisplayAvatar }
+						/>
+
+						<ToggleControl
+							label={ __( 'Display timestamp' ) }
+							checked={ displayTimestamp }
+							onChange={ this.toggleDisplayTimestamp }
+						/>
+
+						<ToggleControl
+							label={ __( 'Display excerpt' ) }
+							checked={ this.props.attributes.displayExcerpt }
+							onChange={ this.toggleDisplayExcerpt }
+						/>
+
+						<TextControl
+							label={ __( 'Number of comments to show' ) }
+							type="number"
+							min={ MIN_COMMENTS }
+							max={ MAX_COMMENTS }
+							value={ this.props.attributes.commentsToShow }
+							onChange={ ( value ) => this.changeCommentsToShow( value ) }
+						/>
+					</InspectorControls>
+				),
+				<ul className={ this.props.className } key="latest-comments">
+					{ latestComments.map( ( comment, i ) =>
+						<li key={ i }>
+							{ displayAvatar && comment.author_avatar_urls[ 96 ] &&
+								<img className={ `${ this.props.className }__comment-avatar` } alt={ comment.author_name } src={ comment.author_avatar_urls[ 96 ] } />
+							}
+							<a href={ comment.link } target="_blank">{ comment._embedded.up[ 0 ].title.rendered }</a>
+							{ displayTimestamp && comment.date_gmt &&
+								<span className={ `${ this.props.className }__comment-timestamp` }>
+									{ moment( comment.date_gmt ).local().format( 'MMM DD h:mm A' ) }
```

Comment:
Moment.js does support relative time: https://momentjs.com/docs/#/displaying/fromnow/

It could be a nice block attribute to decide whether the dates are shown in relative time or absolute time, according to the site's datetime format. 

See also https://github.com/WordPress/gutenberg/pull/1992#issuecomment-317650041 and https://github.com/WordPress/gutenberg/pull/1992#issuecomment-317657482